libs/scalalib/src/mill/scalalib/ScalaModule.scala

```diff
@@ -259,6 +259,13 @@ trait ScalaModule extends JavaModule with TestModule.ScalaModuleBase
     super.mandatoryMvnDeps() ++ scalaLibraryMvnDeps()
   }
 
+  override private[mill] def repositoriesTask0 = Task.Anon {
+    super.repositoriesTask0() ++ Seq(
+      // Scala 3 compiler nightlies
+      coursier.maven.MavenRepository("https://repo.scala-lang.org/artifactory/maven-nightlies")
+    )
+  }
```

Comment:
IIUC, this is only needed by users who want to use nighly Scala 3.8 versions, but not once 3.8.0 is final. We should not add it here, but let the user add it if needed.